### PR TITLE
Writing tests for sunpy.cm

### PR DESCRIPTION
**Pull request by swapsha96**
This PR needs assistance.
Issue #210

### CODE REVIEW

**Comment by dpshelio** [comment]
`astropy_helpers` shouldn't be committed (please amend the commit to remove these changes) and there's a couple of suggestions by pep8speaks.

**Comment by swapsha96** [comment]
Thank you, @dpshelio! Unfortunately, I messed up before pushing files. I'll keep this in mind from now on. I made new changes and now, I am looking forward to suggestions for the test file.

**Comment by nabobalis** [comment]
Thinking of tests for this, I guess we would want to make sure that the SunPy maps are imported into the matplotlib colormap namespace. Also that `get_cmap` returns the correct colour maps.

File: sunpy/cm/tests/test_cm.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import sunpy.cm
+
+import pytest
+import sunpy.cm as cm
+import matplotlib.pyplot as plt
+
```
Comment (nabobalis) [inline sunpy/cm/tests/test_cm.py]: Import for sunpy.cm is repeated. 

I think the order should be 

```
import pytest
import matplotlib.pyplot as plt

import sunpy.cm as cm
```

File: sunpy/cm/tests/test_cm.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def test_get_cmap():
+    colormap_list = cm.cmlist
+    for cmap in colormap_list.keys():
+        assert colormap_list[cmap] == cm.get_cmap(cmap)
+
+
+def test_show_colormaps():
+    if cm.show_colormaps() is None:
+        assert True
```
Comment (nabobalis) [inline sunpy/cm/tests/test_cm.py]: Can you explain this test to me?

File: sunpy/cm/tests/test_cm.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import pytest
+import sunpy.cm as cm
+import matplotlib.pyplot as plt
+
+
+def test_get_cmap():
+    colormap_list = cm.cmlist
+    for cmap in colormap_list.keys():
+        assert colormap_list[cmap] == cm.get_cmap(cmap)
```
Comment (nabobalis) [inline sunpy/cm/tests/test_cm.py]: I think we should test the ```plt.get_cmap``` instead of SunPy.

File: sunpy/cm/tests/test_cm.py (hunk 2 above)
Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: Actually, the issue #210 doesn't contain details, so I was waiting for suggestions to test the function. I know this snippet is testing nothing. Could you please suggest what precisely needs to be checked?

Comment (nabobalis) [reply sunpy/cm/tests/test_cm.py]: Well I am unsure how you would test a show colour map without making it a figure we test the hash against.

Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: @dpshelio @Cadair

File: sunpy/cm/tests/test_cm.py (hunk 1 above)
Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: Fixed it.

File: sunpy/cm/tests/test_cm.py (hunk 3 above)
Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: Fixed it.

File: sunpy/cm/tests/test_cm.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/cm/tests/test_cm.py]: You should create this as a figure test.

**Comment by nabobalis** [comment]
We should depreciate the SunPy's get_cmap function. It is not needed anymore.

File: sunpy/cm/tests/test_cm.py (hunk 2 above)
Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: @nabobalis Are you suggesting to to create a test figure using matplotlib and then to image comparision?

Comment (nabobalis) [reply sunpy/cm/tests/test_cm.py]: Yes

Comment (nabobalis) [reply sunpy/cm/tests/test_cm.py]: It might be alot of effort to make it.

Comment (swapsha96) [reply sunpy/cm/tests/test_cm.py]: @nabobalis I'll try my best. Would let you know if any help is needed.

**Comment by nabobalis** [comment]
@swapsha96 Any chance of an update on this?

**Comment by wafels** [comment]
This looks OK.

**Pull request merged**